### PR TITLE
Feature/try to implement extract checksum

### PR DESCRIPTION
**Pull request by DadouBordeaux**
Required to implement #3

### CODE REVIEW

**Comment by DadouBordeaux** [comment]
Oops, I just wanted to show you the extract_checksum function... but it has taken several commits

**Comment by narfai** [comment]
Taking the filename without extension is not enough, the formatting is more complex, you need also to split with `_`
static-package-a_0.0.1_**ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad**_f61f10025ad.packster

To simplify your code you could use :
```rust
let filename_without_extension = path.file_stem();
```

File: packster-core/src/operation/init_location.rs [hunk 1]
```diff
@@ -42,8 +42,8 @@ impl InitLocationOperation<New> {
     }
 }
 
-pub fn abort_if_something_already_present<F: FileSystem, P: AsRef<Path>>(path_buff: P, filesystem: &F) -> Result<()> {
-    //TODO check if the location is not already present        
+pub fn abort_if_something_already_present<F: FileSystem, P: AsRef<Path>>(_path_buff: P, _filesystem: &F) -> Result<()> {
+    //TODO check if the location is not already present
     Ok(())
```
Comment (DadouBordeaux) [inline packster-core/src/operation/init_location.rs:47]: You could maybe use some logging ? To just print something if there is something in the location.

File: packster-core/src/operation/init_location.rs [hunk 2]
```diff
@@ -58,47 +58,44 @@ pub fn abort_if_something_already_present<F: FileSystem, P: AsRef<Path>>(path_bu
...
-
 pub struct PackagedNotYetInstalled;
 
 
 
 pub fn install_package_to_location<P: AsRef<Path>, L: AsRef<Path>, F: FileSystem, A: Archiver, S: Serializer>(path_package: P, path_location: L, filesystem: &F, archiver: &A, serializer: &S) -> Result<()> {
     // get checksum from package
-    let checksum = extract_checksum(path_package.as_ref())?;
-    // create deploy path from checksum to deploy package 
+    let package = Package::from_path(path_package.as_ref());
```
Comment (DadouBordeaux) [inline packster-core/src/operation/init_location.rs:64]: I think that package is not used, you could do 
```
let checksum = Package::from_path(path_package.as_ref()).to_checksum()

File: packster-core/src/operation/init_location.rs (hunk 1 above)
Comment (DadouBordeaux) [reply packster-core/src/operation/init_location.rs:47]: Something that is bothering me is that we could forgot to handle this case

Comment (narfai) [reply packster-core/src/operation/init_location.rs:47]: I usually solve all TODOs ( or tranform them into issues ) before releases, but ok

**Review by narfai — APPROVED** [review]
Looks good to me !
Some cleanup needed in init_location, next time we should better split activities into dedicated branches.

**Pull request merged**